packages/experience/src/hooks/use-start-webauthn-processing.ts

```diff
@@ -0,0 +1,50 @@
+import { MfaFactor } from '@logto/schemas';
+import { useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import {
+  createWebAuthnRegistrationOptions,
+  generateWebAuthnAuthnOptions,
+} from '@/apis/interaction';
+import { UserMfaFlow } from '@/types';
+import { type WebAuthnState, type MfaFlowState } from '@/types/guard';
+
+import useApi from './use-api';
+import useErrorHandler from './use-error-handler';
+
+type Options = {
+  replace?: boolean;
+};
+
+const useStartWebAuthnProcessing = ({ replace }: Options = {}) => {
+  const navigate = useNavigate();
+  const asyncCreateRegistrationOptions = useApi(createWebAuthnRegistrationOptions);
+  const asyncGenerateAuthnOptions = useApi(generateWebAuthnAuthnOptions);
+  const handleError = useErrorHandler();
+
+  return useCallback(
+    async (flow: UserMfaFlow, flowState: MfaFlowState) => {
```

Comment:
`flow` could be a hook input parameters? Looks like the flow won't change once the hook is initiated.

Reply:
This hook is used in the `useMfaErrorHandler` hook, and the flow is not determind until we call this function.